### PR TITLE
Port to Qt6

### PR DESCRIPTION
**Pull request by isf63**
Openbox is considered legacy, but some people may want fewer Qt5 dependencies. I was unsure about the X11 bits but it compiles and works.

### CODE REVIEW

**Comment by 0323pin** [comment]
@isf63 Thank you for this. I've just used it to extend support on NetBSD.

**Comment by stefonarch** [comment]
Had a look into it, it compiles and looks working, but the warnings should be silenced still.

```
/home/stef/git/obconf-qt/src/maindialog.cpp: In destructor ‘virtual Obconf::MainDialog::~MainDialog()’:
/home/stef/git/obconf-qt/src/maindialog.cpp:61:28: warning: cast between incompatible function types from ‘void (*)(gpointer)’ {aka ‘void (*)(void*)’} to ‘GFunc’ {aka ‘void (*)(void*, void*)’} [-Wcast-function-type]
   61 |     g_list_foreach(themes, (GFunc)g_free, NULL);
      |                            ^~~~~~~~~~~~~
/home/stef/git/obconf-qt/src/obconf-qt.cpp: In function ‘void obconf_error(gchar*, gboolean)’:
/home/stef/git/obconf-qt/src/obconf-qt.cpp:50:40: warning: unused parameter ‘modal’ [-Wunused-parameter]
   50 | void obconf_error(gchar* msg, gboolean modal) {
      |                               ~~~~~~~~~^~~~~
[ 81%] Building CXX object src/CMakeFiles/obconf-qt.dir/windows.cpp.o
[ 83%] Building CXX object src/CMakeFiles/obconf-qt.dir/mouse.cpp.o
[ 85%] Building CXX object src/CMakeFiles/obconf-qt.dir/moveresize.cpp.o
[ 86%] Building CXX object src/CMakeFiles/obconf-qt.dir/margins.cpp.o
/home/stef/git/obconf-qt/src/windows.cpp: In member function ‘void Obconf::MainDialog::on_primary_monitor_popup_currentIndexChanged(int)’:
/home/stef/git/obconf-qt/src/windows.cpp:96:12: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
```
and else.

**Comment by isf63** [comment]
@stefonarch 

I silenced the warnings related to unused static functions, unused parameters, and integer sign comparison:
- unused static functions -> comment out
- unused parameters -> add dummy statements, e.g. `(void)(index); // unused`
- integer sign comparisons -> cast to same sign

The `GFunc` warnings are difficult to squash and exist elsewhere, i.e. in libfm-qt.

----

What should be done about `qApp->nativeInterface<QNativeInterface::QX11Application>()` when not running on X11? I was thinking essentially doing `exit(1)`-the-Qt-way...

**Comment by isf63** [comment]
Done.

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,25 +13,24 @@ include(GNUInstallDirs)
...
-set(QT_MINIMUM_VERSION "5.12.0")
+set(LXQTBT_MINIMUM_VERSION "2.0.0")
+set(QT_MINIMUM_VERSION "6.6.0")
 
-set(OBCONFQT_MAJOR_VERSION 0)
-set(OBCONFQT_MINOR_VERSION 16)
-set(OBCONFQT_PATCH_VERSION 4)
+set(OBCONFQT_MAJOR_VERSION 1)
+set(OBCONFQT_MINOR_VERSION 0)
+set(OBCONFQT_PATCH_VERSION 0)
```
Comment (tsujan) [inline CMakeLists.txt:21]: Please only change the patch version to 5, such that it becomes 0.16.5. We don't want to pretend that we really maintain this app.

**Review by tsujan — APPROVED** [review]
The code seems OK.

@stefonarch 
I didn't test it under X11. If it passes your tests, please merge it.

**Comment by stefonarch** [comment]
GTM
Works fine, thanks!

**Comment by stefonarch** [comment]
I think we could release it but better after the main translations came in for this new message.

immagine

**Comment by 0323pin** [comment]
Yeah, I get that. Thanks!

**Comment by tsujan** [comment]
@stefonarch 
Ping me when it's ready to be released.

**Pull request merged**